### PR TITLE
Implement SuperGLUE tasks and baselines

### PR DESCRIPTION
**Pull request by W4ngatang**
WIP implementation of the SuperGLUE tasks, following the style of the GLUE data processors. This PR includes:
- classification with marked spans: most tasks are classification and straightforward, but WiC and WSC have special spans. This PR implements `*SpanClassification` models (currently for just BERT and RoBERTa) and relevant data classes
- scripts for using and extracting results from the [experiment impact tracker](https://github.com/Breakend/experiment-impact-tracker) library, which tracks energy usage and is being used for the SustaiNLP 2020 competition

TODO
- clean up some of my development scripts and cruft

### CODE REVIEW

**Comment by julien-c** [comment]
Hi @W4ngatang did you see that we have a new `Trainer` class in the library? It's basically a refactor of the previous version of `run_glue.py` so should be pretty easy to adopt.

Let us know if we can help, and very excited to have an easy path to SuperGLUE tasks in the repo :)

**Comment by W4ngatang** [comment]
Hi Julien,

Saw it! I started writing this a while ago, but you all move very fast. We're preparing this as some reference code for a competition, so we may keep the training code as is for a bit and then switch to the trainer when we're ready to actually merge into the main repo.

**Comment by W4ngatang** [comment]
@thomwolf

**Comment by leondz** [comment]
Looks super-cool, especially with SustaiNLP just around the corner! Is there any help that could be given that would accelerate this PR?

File: src/transformers/data/processors/superglue.py [hunk 1]
```diff
@@ -0,0 +1,942 @@
...
+        idx2label = {i: label for i, label in enumerate(self.get_labels())}
+        with open(os.path.join(out_dir, "AX-g.jsonl"), "w") as pred_fh:
+            for idx, pred in enumerate(preds):
+                pred_label = idx2label[int(pred)]
+                pred_fh.write(f"{json.dumps({'idx': idx, 'label': pred_label})}\n")
+        logger.info(f"Wrote {len(preds)} predictions to {out_dir}.")
+
+
+superglue_tasks_num_labels = {
+    "ax-b": 2,
```
Comment (Am1n3e) [inline src/transformers/data/processors/superglue.py:892]: I think that `multirc: 2,` is missing here.

**Comment by Tiiiger** [comment]
hi @W4ngatang @thomwolf 

any help you need for this PR? I think pushing this PR through can help many researchers with their experiments.

**Comment by W4ngatang** [comment]
Hi Tianyi,

Yes, help would be great! The code is correct from my end, but uses an old
version of Transformers. At this point, it's just a matter of merging the
most recent version of Transformers in and adhering to the Transformers
contributor guidelines.

Best,
Alex

**Pull request closed without merging**